File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution{
    public int minimumPushes(String w){
        int n=w.length();

        int[] mp=new int[26];
        for(char ch: w.toCharArray()){
            int idx=ch-'a';
            mp[idx]++;
        }

        int[] sorted = Arrays.stream(mp)
                        .boxed()
                        .sorted(Collections.reverseOrder())
                        .mapToInt(Integer::intValue)
                        .toArray();
        
        int t=0;
        int res=0;
        for(int x: sorted){
            if(x == 0)
                break;
            int p=t/8+1;
            res += (p*x);
            t++;
        }

        return res;


    }
}

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, w: str) -> int:
        n=len(w)

        mp=[0]*26
        for ch in w:
            idx=ord(ch)-ord('a')
            mp[idx]+=1

        sortc=sorted(mp, reverse=True)
        
        res=0
        for idx, count in enumerate(sortc):
            if count == 0:
                break
            res += (idx//8+1)*count

        return res

